dtfs-central-api/api-tests/v1/tfm/deal-cancellation/tfm-deal-cancellation-post.api-test.ts

```diff
@@ -0,0 +1,156 @@
+  });
+
+  afterEach(async () => {
+    process.env = { ...originalProcessEnv };
+    await wipeDB.wipe([MONGO_DB_COLLECTIONS.TFM_DEALS, MONGO_DB_COLLECTIONS.DEALS]);
+  });
+
+  describe('POST /v1/tfm/deals/:dealId/cancellation/submit', () => {
+    describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is disabled', () => {
+      beforeEach(() => {
+        process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'false';
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+      });
+
+      it('should return 404', async () => {
+        const { status } = await testApi.post({ cancellation, auditDetails }).to(submitDealCancellationUrl);
+
+        expect(status).toEqual(HttpStatusCode.NotFound);
+      });
+    });
+
+    describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is enabled', () => {
+      beforeEach(() => {
+        process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'true';
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+      });
+
+      withMongoIdPathParameterValidationTests({
+        baseUrl: '/v1/tfm/deals/:dealId/cancellation/submit',
+        makeRequest: (url) => testApi.post({ cancellation, auditDetails }).to(url),
+      });
+
+      it('should return the submit cancellation response object if a matching deal and cancellation exists', async () => {
+        const submitCancellationResponse = await testApi.post({ cancellation, auditDetails }).to(submitDealCancellationUrl);
+
+        expect(submitCancellationResponse.body).toEqual({ cancelledDeal: { ukefDealId: dealId } });
+        expect(submitCancellationResponse.status).toEqual(HttpStatusCode.Ok);
+      });
+
+      it('should return 404 if deal is valid but the existing cancellation does not match the passed in params', async () => {
+        const differentDealCancellationDetails = { ...cancellation, reason: 'a different reason' };
+
+        const submitCancellationResponse = await testApi.post({ cancellation: differentDealCancellationDetails, auditDetails }).to(submitDealCancellationUrl);
+
+        expect(submitCancellationResponse.status).toEqual(HttpStatusCode.NotFound);
+        expect(submitCancellationResponse.body).toEqual({
+          status: HttpStatusCode.NotFound,
+          message: `Deal not found: ${dealId}`,
+        });
+      });
+
+      it('should return 404 if dealId is valid but not associated to a record', async () => {
+        const validButNonExistentDealId = new ObjectId();
+
+        const submitCancellationResponse = await testApi
+          .post({ cancellation, auditDetails })
+          .to(`/v1/tfm/deals/${validButNonExistentDealId.toString()}/cancellation/submit`);
+
+        expect(submitCancellationResponse.status).toEqual(HttpStatusCode.NotFound);
+        expect(submitCancellationResponse.body).toEqual({
+          status: HttpStatusCode.NotFound,
+          message: `Deal not found: ${validButNonExistentDealId.toString()}`,
+        });
+      });
+
+      it('should return 400 if invalid dealId', async () => {
```

Comment:
do we want payload validation tests too - is there any middleware checking the body?

Reply:
I've got payload validation tests here: https://github.com/UK-Export-Finance/dtfs2/pull/3696/files/3e74e750a55f4fb1f6dd401f8301f69a8ac09c75#diff-3656c6d081296b38b6dea4ce92e41c0dd8fc9deaf345556b57bdf7082499c578

Reply:
Its probably worth adding something in the unit tests just to check its all wired up correctly - doesnt have to be anythign exhaustive just maybe 'returns 400 when the payload is invalid' and test one case

Reply:
Ah I'm with you - have now added! :)